File: leetcode/strings/0058_length_of_last_word.cpp
// LC 58. Length of Last Word | Easy
#include <string>
using namespace std;

class Solution {
public:
    int lengthOfLastWord(string s) {
        int i = s.size() - 1, len = 0;
        while (i >= 0 && s[i] == ' ') i--;
        while (i >= 0 && s[i] != ' ') { len++; i--; }
        return len;
    }
};

File: leetcode/strings/0058_length_of_last_word.py
# LC 58. Length of Last Word | Easy
class Solution:
    def lengthOfLastWord_pythonic(self, s):
        return len(s.split()[-1])

    def lengthOfLastWord(self, s):
        i = len(s) - 1
        while i >= 0 and s[i] == ' ': i -= 1
        j = i
        while j >= 0 and s[j] != ' ': j -= 1
        return i - j
